OvmfPkg/IntelTdx/TdxHelperLib/SecTdxHelper.c

```diff
@@ -523,6 +523,467 @@ IsInValidList (
+  @retval  EFI_INVALID_PARAMETER      TdxMetadataAddress or UnacceptedMemoryLength was invalid.
+  @retval  EFI_NOT_FOUND              Tdx Metadata info was not found.
+  @retval  EFI_UNSUPPORTED            We don't support the unaccepted memory address,
+                                      that was in the range of Tdx Metadata Temporay Memory.
+**/
+STATIC
+EFI_STATUS
+CheckUnacceptedMemory (
+  IN UINT8   *TdxMetadataAddress,
+  IN UINT64  UnacceptedMemoryStart,
+  IN UINT64  UnacceptedMemoryLength
+  )
+{
+  UINTN       Index;
+  UINTN       FlagErrorMessage;
+  UINT64      MemoryOffset;
+  UINT8       *Address;
+  EFI_STATUS  Status;
+
+  TDVF_METADATA_DESCRIPTOR  *TdvfDec;
+  TDX_METADATA_SECTION      *Section;
+
+  MemoryOffset     = 0;
+  FlagErrorMessage = 0;
+  Status           = EFI_SUCCESS;
+  Address          = NULL;
+  TdvfDec          = NULL;
+  Section          = NULL;
+
+  if (TdxMetadataAddress == NULL) {
+    DEBUG ((DEBUG_ERROR, "TdxMetadataAddress should not be NULL\n"));
+    Status = EFI_INVALID_PARAMETER;
+    return Status;
+  }
+
+  if (UnacceptedMemoryLength == 0) {
+    DEBUG ((DEBUG_ERROR, "UnacceptedMemoryLength should not be zero\n"));
+    Status = EFI_INVALID_PARAMETER;
+    return Status;
+  }
+
+  Address = TdxMetadataAddress;
+  TdvfDec = (TDVF_METADATA_DESCRIPTOR *)Address;
+
+  Address += sizeof (TDVF_METADATA_DESCRIPTOR);
+
+  for (Index = 0; Index < TdvfDec->NumberOfSectionEntry; Index++) {
+    Section = (TDX_METADATA_SECTION *)Address;
+
+    if (Section->Type != TDX_METADATA_SECTION_TYPE_TEMP_MEM) {
+      Address += sizeof (TDX_METADATA_SECTION);
+      Section  = NULL;
+      continue;
+    }
+
+    if (UnacceptedMemoryStart == Section->MemoryAddress) {
+      FlagErrorMessage = 1;
+      break;
+    }
+
+    if (UnacceptedMemoryStart > Section->MemoryAddress) {
+      MemoryOffset = UnacceptedMemoryStart - Section->MemoryAddress;
+      if (MemoryOffset < Section->MemoryDataSize) {
+        FlagErrorMessage = 1;
+        break;
+      }
+    } else {
+      MemoryOffset = Section->MemoryAddress - UnacceptedMemoryStart;
+      if (MemoryOffset < UnacceptedMemoryLength) {
+        FlagErrorMessage = 1;
+        break;
+      }
+    }
+
+    Address += sizeof (TDX_METADATA_SECTION);
+    Section  = NULL;
+  }
+
+  if (FlagErrorMessage == 1) {
+    DEBUG ((DEBUG_ERROR, "Unaccepted Memory should not be in the range of Temporay Memory\n"));
+    Status = EFI_UNSUPPORTED;
+    return Status;
+  }
+
+  return Status;
+}
+
+/**
+  Gets the TDX Metadata Offset by the BFV Base Address and the Chain footer struct.
+  Chain footer struct was defined on the Reset Vector
+  (OvmfPkg/ResetVector/Ia16/ResetVectorVtf0.asm).
+
+  @param[in]      BFVBaseAddress    The address of the BFV
+  @param[in,out]  TdxMedataAddress  The address of the Tdx Metadata if found
+
+  @retval  EFI_SUCCESS              BFVBaseAddress was valid and the address of the Tdx Metadata was found.

```

Comment:
@retval  EFI_SUCCESS              Address of the Tdx Metadata was found successfully.